CMakeLists.txt

```diff
@@ -269,7 +269,13 @@ execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
 
 # We need to link some packages as dynamic/dependent.
 set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
+if(APPLE)
+  execute_process(COMMAND brew --prefix OUTPUT_VARIABLE BREW_PREFIX)
+  set(OPENSSL_ROOT_DIR "${BREW_PREFIX}/opt/openssl")
```

Comment:
The default build should be more or less 'deployable', so no brew-provided dependent dynamic linking. We could provide the builder with 2 generic options: (1) Use the system-provided OpenSSL and link dynamically; (2) Use brew-provided and link statically. Off-the-cuff tests of a static link show about 1.5M of added code, not good, but tolerable.

We should allow a third option, linking brew-provided OpenSSL dynamically: when the builder sets the `BUILD_LINK_SHARED` environment variable. Thoughts?